src/Generator/EntityGenerator.php

```diff
@@ -485,9 +494,39 @@ private function generateMainEntityData(\SimpleXMLElement $element, &$relationVa
             }
         }
 
+        if (isset($this->entityModel['fields']['unique'])) {
+            $uniqueFields = explode(',', $this->entityModel['fields']['unique']);
+            $key = [];
+            foreach ($uniqueFields as $uniqueField) {
+                $key[] = (string) $child->attributes()[$uniqueField];
+            }
+            $key = implode('-', $key);
+
+            if (in_array($key, $this->existingChildKeys)) {
+                $child = $this->generateMainEntityData($element, $relationValues);
```

Comment:
If I get it right here, you recursively call `generateMainEntityData` until you somehow reach randomly a child which doesn't match the existing keys. Be careful with this kind of algorithm you can end up very easily in a infinite loop just because out of bad luck you never find a unique key. Or even worse sometimes you simply can't based on the max allowed possibilities from the set of attributes.

Reply:
Now implementing a stable alternative may turn out very complex (you'd need to keep track of used attributes while you generate them for example), so maybe it's not worth too much complexity.